transport/src/main/java/io/netty/channel/pool/SimpleChannelPool.java

```diff
@@ -219,7 +238,7 @@ public void run() {
         return promise;
     }
 
-    private void doReleaseChannel(Channel channel, Promise<Void> promise) {
+    private void doReleaseChannel(final Channel channel, final Promise<Void> promise) {
```

Comment:
Lets revert for this PR. I understand the rational, and do not necessarily disagree, but lets keep the set of changes focused on what is required to resolve the issue.  If need be we can open a followup PR focused on using `final` more consistently.